packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -163,6 +161,25 @@ class CodeSnippetDisplay extends MetadataDisplay<
     }
   };
 
+  // Verify if a given widget is a FileEditor
+  private isFileEditor = (widget: Widget): boolean => {
+    return (
+      widget instanceof DocumentWidget &&
+      (widget as DocumentWidget).content instanceof FileEditor
+    );
+  };
+
+  // Return the code editor from a given widget
+  private getFileEditor = (widget: Widget): CodeEditor.IEditor => {
+    const documentWidget = widget as DocumentWidget;
+    return (documentWidget.content as FileEditor).editor;
+  };
```

Comment:
You should be able to turn this into a type guard:
```ts
// type guard
const isFileEditor = (widget: Widget): widget is DocumentWidget<FileEditor> => {
  return (widget as DocumentWidget).content instanceof FileEditor;
};


if (isFileEditor(widget)) {
  // the widget should be automatically typed properly inside the if statement
  const fileEditor = widget.content.editor;
}
```

Reply:
Thank you @bourdakos1 , I like how cleaner it looks now